app/models/activity.rb

```diff
@@ -49,8 +56,16 @@ class Activity < ApplicationRecord
     wizard_status == "dates" || wizard_complete?
   end
 
-  private def country_step?
-    wizard_status == "country" || wizard_complete?
+  def geography_step?
+    wizard_status == "geography" || wizard_complete?
+  end
+
+  def region_step?
+    wizard_status == "region"
+  end
+
+  def country_step?
+    wizard_status == "country"
```

Comment:
Yeah, I've tried to cover this in the PR message, but obviously failed! :)

`wizard_complete?` is true when the user has completed all of the steps in the wizard.

If the user supplies a region and not a country, at the point the wizard status is complete the activity is no longer valid as it has no recipient_country.

I decided to loosen the validation to run only on the step collecting the value for the two branched attributes, so whichever attribute they chose they must supply a value.

Does that make sense? I am struggling to explain this in writing!?

Reply:
Ok. I think that makes sense. I think what confuses me is the name of the methods in here. Like why is `wizard_complete?` being checked in tests to see if we're on a step? Would it be clearer to refactor this to run validations on multiple conditions: `*_step?` and `wizard_complete?`?

Reply:
I am not sure it would be clearer, but I do agree the whole step/which step/complete/valid setup needs more work - I was not convinced this was the place to do it though?

Reply:
I think you might be right that this isn't the right place to do it, but wouldn't `if: [:*_step?, :wizard_complete?]` (I think that's valid?) be clearer?